Sources/SwiftDocC/Semantics/Metadata/Metadata.swift

```diff
@@ -132,6 +137,107 @@ public final class Metadata: Semantic, AutomaticDirectiveConvertible {
+            categorizedAvailability[availability.platform, default: []].append(availability)
+        }
+
+        for availabilityAttrs in categorizedAvailability.values {
+            guard availabilityAttrs.count > 1 else {
+                continue
+            }
+
+            let duplicateIntroduced = availabilityAttrs.filter({ $0.introduced != nil })
+            let duplicateBeta = availabilityAttrs.filter({ $0.isBeta })
+            let duplicateDeprecated = availabilityAttrs.filter({ $0.isDeprecated })
+            if duplicateIntroduced.count > 1 {
+                for avail in duplicateIntroduced {
+                    let diagnostic = Diagnostic(
+                        source: avail.originalMarkup.nameLocation?.source,
+                        severity: .warning,
+                        range: avail.originalMarkup.range,
+                        identifier: "org.swift.docc.\(MetadataAvailability.self).DuplicateIntroduced",
+                        summary: "Duplicate \(MetadataAvailability.directiveName.singleQuoted) directive with `introduced` argument",
+                        explanation: """
+                        A documentation page can only contain a single `introduced` version for each platform.
+                        """
+                    )
+
+                    guard let range = avail.originalMarkup.range else {
+                        problems.append(Problem(diagnostic: diagnostic))
+                        continue
+                    }
+
+                    let solution = Solution(
+                        summary: "Remove extraneous \(MetadataAvailability.directiveName.singleQuoted) directive",
+                        replacements: [
+                            Replacement(range: range, replacement: "")
+                        ]
+                    )
+
+                    problems.append(Problem(diagnostic: diagnostic, possibleSolutions: [solution]))
+                }
+            }
+
+            if duplicateBeta.count > 1 {
+                for avail in duplicateBeta {
+                    let diagnostic = Diagnostic(
+                        source: avail.originalMarkup.nameLocation?.source,
+                        severity: .warning,
+                        range: avail.originalMarkup.range,
+                        identifier: "org.swift.docc.\(MetadataAvailability.self).DuplicateBeta",
+                        summary: "Duplicate \(MetadataAvailability.directiveName.singleQuoted) directive with `isBeta` argument",
+                        explanation: """
+                        A documentation page can only be declared `isBeta` once per platform.
+                        """
+                    )
+
+                    guard let range = avail.originalMarkup.range else {
+                        problems.append(Problem(diagnostic: diagnostic))
+                        continue
+                    }
+
+                    let solution = Solution(
```

Comment:
Nice. Thanks for adding more diagnostics with solutions and replacements.

Reply:
This is absolutely copy/pasted from @ethan-kusters' "duplicate page image" warnings 😅